src/modules/musig/musig.md

```diff
@@ -0,0 +1,196 @@
+   - the public key hash output from `secp256k1_musig_pubkey_combine`
+   - the signer's index `i` `my_index`
+   - the signer's secret key `seckey`
+2. The signer then communicates `H(R_i)` to all other signers, and receives
+   commitments `H(R_j)` from all other signers `j`. These hashes are simply
+   length-32 byte arrays which can be communicated however is communicated.
+3. Once all signers nonce commitments have been received, the signer records
+   these commitments with the function `secp256k1_musig_session_get_public_nonce`.
+   This function updates in place
+   - the session state `session`
+   - the array of signer data `signers`
+   taking in as input the list of commitments `commitments` and outputting the
+   signer's public nonce `R_i` in the out-pointer `nonce`.
+4. The signer then communicates `R_i` to all other signers, and receives `R_j`
+   from each signer `j`. On receipt of a nonce `R_j` he calls the function
+   `secp256k1_musig_set_nonce` to record this fact. This function checks that
+   the received nonce is consistent with the previously-received nonce and will
+   return 0 in this case. The signer must also call this function with his own
+   nonce and his own index `i`.
+   These nonces `R_i` are secp256k1 public keys; they should be serialized using
+   `secp256k1_ec_pubkey_serialize` and parsed with `secp256k1_ec_pubkey_parse`.
+5. Once all nonces have been exchanged in this way, signers are able to compute
+   their partial signatures. They do so by calling `secp256k1_musig_session_combine_nonces`
+   which updates in place
+   - the session state `session`
+   - the array of signer data `signers`
+   It outputs an auxilary integer `nonce_is_negated` and has an auxilary input
+   `adaptor`. Both of these may be set to NULL for ordinary signing purposes.
+   If the signer did not provide a message to `secp256k1_musig_session_initialize`,
+   a message must be provided now by calling `secp256k1_musig_session_set_msg` which
+   updates the session state in place.
+6. The signer computes a partial signature `s_i` using the function
+   `secp256k1_musig_partial_sign` which takes the session state as input and
+   partial signature as output.
+7. The signer then communicates the partial signature `s_i` to all other signers, or
+   to a central coordinator. These partial signatures should be serialized using
+   `musig_partial_signature_serialize` and parsed using `musig_partial_signature_parse`.
+8. Each signer calls `secp256k1_musig_partial_sig_verify` on the other signers' partial
+   signatures to verify their correctness. If only the validity of the final signature
+   is important, not assigning blame, this step can be skipped.
+9. Any signer, or central coordinator, may combine the partial signatures to obtain
+   a complete signature using `secp256k1_musig_partial_sig_combine`. This function takes
+   a signing session and array of MuSig partial signatures, and outputs a single
+   Schnorr signature.
```

Comment:
Just a small question: why is there no function for session closing / destruction?

Reply:
Oh, yes, we should add one that zeroes out secret fields. I think we just overlooked this.